stripe/service.go

```diff
@@ -270,14 +289,27 @@ func (s *Service) CreateCheckoutSessionWithLookupKey(
 		return nil, NewStripeError("org_not_found", fmt.Sprintf("organization %s not found", orgAddress), err)
 	}
 
+	// Determine if the organization is eligible for a free trial based on
+	// if they already made a subscription and request a yearly plan
+	freeTrialEligible := org.Subscription.LastPaymentDate.IsZero() && billingPeriod == string(db.BillingPeriodAnnual)
```

Comment:
why another name for the same concept? could be hasFreeTrial
but my suggestion would be to turn this into 
```go
if org.Subscription.LastPaymentDate.IsZero() && billingPeriod == string(db.BillingPeriodAnnual) {
  params.TrialPeriodDays = s.config.FreeTrialDays
}
```
after the `params := &CheckoutSessionParams`